src/main/java/com/lithium/dbi/rdbi/recipes/locking/MultiReadWriteLock.java

```diff
@@ -0,0 +1,90 @@
+package com.lithium.dbi.rdbi.recipes.locking;
+
+import com.google.common.base.Throwables;
+import com.lithium.dbi.rdbi.Handle;
+import com.lithium.dbi.rdbi.RDBI;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+
+public class MultiReadWriteLock {
+
+    private final RDBI rdbi;
+    private final String writeLockKey;
+    private final String readLockKey;
+    private final int writeTimeoutMillis;
+    private final int readTimeoutMillis;
+
+    public MultiReadWriteLock(RDBI rdbi,
+                              String writeLockKey,
+                              String readLockKey,
+                              int writeTimeoutMillis,
+                              int readTimeoutMillis) {
+        this.rdbi = rdbi;
+        this.writeLockKey = writeLockKey;
+        this.readLockKey = readLockKey;
+        this.writeTimeoutMillis = writeTimeoutMillis;
+        this.readTimeoutMillis = readTimeoutMillis;
+    }
+
+    public boolean acquireWriteLock(String ownerId) {
+        try (Handle handle = rdbi.open()) {
+            final MultiReadWriteLockDAO dao = handle.attach(MultiReadWriteLockDAO.class);
+            while (true) {
+                final int acquiredLock = dao.acquireWriteLock(writeLockKey, ownerId, writeTimeoutMillis);
+                if (acquiredLock > 0) {
+                    break;
+                }
+
+                Thread.sleep(250);
+            }
+
+            while (true) {
```

Comment:
might be worth adding a comment about what case you are addressing here. id imagine others will wonder why you acquire the write lock and then wait for the read lock to get acquired like i did